### PR TITLE
Invalidate CloudFront when publishing

### PR DESCRIPTION
**Pull request by dbnicholson**
This converts the publishing script to python so it can keep track of which objects were updated and then invalidate them in the CloudFront CDN. You could also do this in a more brute force way by invalidating the entire CDN on every update, but I had something like this lying around from a similar project.

https://phabricator.endlessm.com/T31257

### CODE REVIEW

File: publish-docs.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
```
Comment (pwithnall) [inline publish-docs.py:1]: Ideally would have a license header.

File: publish-docs.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    # Create the CloudFront invalidation.
+    logger.info(f'Invalidating CloudFront distribution {args.cloudfront}')
+    if not args.dry_run:
+        resp = cloudfront.create_invalidation(
+            DistributionId=args.cloudfront,
+            InvalidationBatch={
+                'Paths': {
+                    'Quantity': len(paths),
+                    'Items': paths
```
Comment (pwithnall) [inline publish-docs.py:139]:
```suggestion
                    'Items': paths,
```

(nitpick, sorry)

File: publish-docs.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                },
+                'CallerReference': caller_ref
```
Comment (pwithnall) [inline publish-docs.py:141]:
```suggestion
                'CallerReference': caller_ref,
```

(nitpick, sorry)

**Review by pwithnall — APPROVED** [review]
LGTM with/without my awfully pedantic nitpicks

File: publish-docs.py (hunk 1 above)
Comment (dbnicholson) [reply publish-docs.py:1]: Good call. I too often start writing the code without thinking about the license.

**Pull request merged**